src/checkstyle/checks.xml

```diff
@@ -32,6 +32,10 @@
         <property name="message" value="Blank line before end of file" />
     </module>
     <module name="RegexpSingleline">
+        <property name="format" value="(class|interface) ([a-zA-Z0-9])* (extends|implements)" />
```

Comment:
I suggest leaving the generics case for simplicity. 

We can also have `(class|interface) ([a-zA-Z0-9])*(<.*>)* (extends|implements)` to catch extends in the same line for generics case up to depth of 1. Or further, but we don't get much by making regex complicated.